### ISSUE TITLE
mongodb-core plugin doesn't work with Mongoose

### ISSUE THREAD

**Issue by AndrewLugg** [DataDog/dd-trace-js#221]
Please can you consider adding a [Mongoose](http://mongoosejs.com/) plugin. 

It is built on top of MongoDB-core, but it doesn't work, it is probably an easy tweak.

**Comment by rochdev** [comment]
What is the version of the underlying `mongodb-core` that is pulled as a transitive dependency?

**Comment by AndrewLugg** [comment]
I tried it with Mongoose 5.2.4, which uses mongodb-core 3.1.0

**Comment by AndrewLugg** [comment]
If the mongodb-core plugin worked with Mongoose that would be great, I don't think a specific mongoose plugin is needed.

**Comment by rochdev** [comment]
Just wrote a simple test case using `findOne()` and I get a span as expected.

Are you loading `mongoose` before or after calling `tracer.trace()`? Also, are you using `import` or `require`? Sometimes the order will change once transpiled when using `import` because of hoisting.

**Comment by AndrewLugg** [comment]
I am using Express, and Mongoose, Express tracing is working fine, and I get the http requests too, but mongo does nothing. 

Here is a copy of the order I am bringing in the my libraries:
```
const when = require('when');
const request = require('request');
const requestRetry = require('requestretry');
const ioredis = require('ioredis');
const compression = require('compression');
const config = require('./src/config.js');
const tracer = require('dd-trace').init({
    service : 'ls-service-'+ config.ENVIRONMENT,
    //hostname: 'service.' + config.ENVIRONMENT,
    hostname: 'host.docker.internal',
    port: 8126,
    debug: true,
});
tracer.use('express');
//tracer.trace();
const mongoose = require('mongoose');
const express = require('express');
const app = express();
const bodyParser = require('body-parser');
```

This is the one I just tested adding tracer.trace(); and I get a deprecatin warning for it saying "Please use tracer.startSpan() and tracer.scopeManager() instead.See: https://datadog.github.io/dd-trace-js/#manual-instrumentation."

I also tried adding tracer.use('mongodb-core'); but from what I understand, it doesn't matter if I have plugins defaulted to true.

**Comment by rochdev** [comment]
It is correct that `tracer.use()` is not necessary if you don't explicitly disable all plugins in the configuration.

The code from the snippet you provided seems correct. Would you be able to provide a small sample of Mongoose code that you would expect to get a span but doesn't?

Also, which version of the tracer are you using?

**Comment by AndrewLugg** [comment]
I decided to try it on a different server that has a more simplified layout. And it is working with Mongoose and Express. The difference is that it has all of its functions in the 1 file, and the model as a separate file. The service it wasn't working with, has the express message handlers in an a separate file, which call the models in.

```
Server1 (not working)
app.js
---> Message-handlers.js
      \---> Models.js

Server2 (working)
app.js
---> Models.js
```

I am using dd-trace version 0.5.3

**Comment by rochdev** [comment]
This shouldn't change the behavior. Is it possible maybe that this line `const config = require('./src/config.js');` requires `mongoose` by indirection?

**Comment by rochdev** [comment]
This was fixed in #241 

@AndrewLugg Can you confirm if it ended up working for you?

**Comment by AndrewLugg** [comment]
Yes this is working for us. Thank you.
